.github/ci_templates/build.yml

```diff
@@ -157,5 +162,13 @@ steps:
         fi;
       fi
 {!{- end }!}
+
+  - name: Save dev/install image name to cache after build
+    if: ${{ env.WERF_ENV == 'FE' }}
+    uses: {!{ index (ds "actions") "actions/cache" }!}
```

Comment:
I did not get the part about caching. AFAIK, you can use the image `${registry}/sys/deckhouse-oss/dev/install:prXXX`, and it should be in the registry after build. 

E2E should not require running build again if the image is in the registry.

Reply:
This caching is an assurance that E2E test uses the image built for the latest commit on the branch. It is not a requirement to build image again, it is a mechanism to prevent running e2e tests before "Build FE" job finishes for latest commit.

On the other hand, using prXXX image is more convenient and caching is not a reliable tool, as it expire in 2 weeks. We already have problems with expired modules_images_json files. Also, werf sets a `werf-project-repo-commit` label on the image, so we can check if the image is related to the current commit and fail the E2E job.

There is one dangerous situation, when "Build FE" and "E2E: provider" jobs run on the same runner and interfere with each other, but we have dedicated runners for e2e.